frontend/src/main/components/Users/UsersTable.js

```diff
@@ -16,13 +21,33 @@ export default function UsersTable({ users }) {
     };
   }
 
-  // Stryker disable all : hard to test for query caching
-  const toggleAdminMutation = useBackendMutation(
-    cellToAxiosParamsToggleAdmin,
-    {},
-    ["/api/admin/users"],
+  // Custom mutation that handles errors without the default Axios error
+  const toggleAdminMutation = useMutation(
+    async (cell) => {
+      try {
+        const params = cellToAxiosParamsToggleAdmin(cell);
+        const response = await axios(params);
+        return response.data;
+      } catch (error) {
+        if (
+          error.response &&
+          error.response.data &&
+          error.response.data.message
+        ) {
+          toast(`Error: ${error.response.data.message}`);
+        } else {
+          toast(`Error: ${error.message || "Unknown error occurred"}`);
+        }
+        throw error; // Re-throw to maintain error state in mutation
+      }
+    },
+    {
+      onSettled: () => {
+        queryClient.invalidateQueries(["/api/admin/users"]);
+      },
+      retry: false,
+    },
```

Comment:
I won't necessarily make you change this, but I'll point out that there is an easier way:

See: https://ucsb-cs156.github.io/topics/spring_react/spring_react_propagating_errors.html